docs/source/miscellaneous/doc/adr/0016-use-tickets-encoding-for-transactions.md

```diff
@@ -499,7 +499,159 @@ the worker pool has a finite size and even with the full batching algorithm in t
+
+Values provided are valid up to a specific timestamp, called the validity bound. When a transform is applied, a fresh
+timestamp ``F`` is taken, and the new value will be written with a validity bound ``F + C`` for some constant ``C``.
+Going forward, the behaviour for updates up to the previous bound must be preserved (which means decisions would be the
+same whether other service nodes read the new value or not). Nodes must not make decisions based on values read with a
+validity bound less than their timestamp of interest.
+
+```java
+public interface CoordinationService<T> {
+    Optional<ValueAndBound<T>> getValueForTimestamp(long timestamp);
+
+    CheckAndSetResult<ValueAndBound<T>> tryTransformCurrentValue(Function<ValueAndBound<T>, T> transform);
+
+    // ONLY FOR METRICS AND MONITORING - NOT FOR PRODUCTION DECISIONS.
+    Optional<ValueAndBound<T>> getLastKnownLocalValue();
+}
+```
+
+#### Physical Implementation
+
+The coordination service needs to persist state; we've implemented this as a non-transactional table in the AtlasDB
+key-value service called ``_coordination``. Within this table, each sequence of values being agreed on takes up one row.
+The values themselves are stored as dynamic columns with a var-long dynamic column key.
+
+- The value stored at column 0 is a ``SequenceAndBound`` which indicates the sequence number of the currently valid
+  value and the validity bound.
+- Values are stored at column IDs equal to a fresh timestamp at the time they're written.
+
+For example, the Cassandra representation of the coordination table may look as follows.
+
+|  key | column1 | column2 |                                                                  value |
+|-----:|--------:|--------:|-----------------------------------------------------------------------:|
+| 0x6d |    0x00 |      -1 | 0x7b2273657175656e6365223a31333030372c22626f756e64223a353031333030377d |
+| 0x6d |  0xa711 |      -1 |                                     0x7b2276657...3334227d (216 bytes) |
+| 0x6d |  0xb2cf |      -1 |                                     0x7b2276657...3930227d (380 bytes) |
+
+The single ``key`` reflects that these values are all part of the same sequence of values. The column IDs are
+``VAR_LONG`` encoded; ``0x00`` maps to zero, and the associated value decodes to a ``SequenceAndBound``, in this
+example ``{"sequence":13007,"bound":5013007}``. This means that the value at the column associated with sequence
+``13007`` (which encodes to ``b2cf``) is the currently agreed value; the value in column ``a711`` is not current.
+
+The main reason for this scheme is that while the validity bound itself changes frequently (approximately every
+``ADVANCEMENT_QUANTUM`` timestamps, which is 5,000,000 at time of writing), the value itself changes much less often.
+For example, with transactions2, in the absence of rollbacks, the value is expected to change just once in the
```

Comment:
decide on whether to backtick `transactions2` or not and then stick to it

Reply:
In my usage, transactions2 shouldn't have backticks unless it is the table specifically. So earlier in the doc we have 

> As far as transactions2 is concerned, we observe that the number of partitions is very small, and thus we can go with a very low setting. We thus set ``bloom_filter_fp_chance`` to 0.0001. Empirically, we observed that for 1,000,000,000 timestamps, the bloom filter we had for the ``_transactions2`` table even with this setting was about 15 KB, while that
for ``_transactions`` was about 580 MB.

As for the phrase "read(ing) from transactions2", that would depend if it means reading values written by the scheme in general (no backticks), or reading values from the table itself (should have backticks). So I'd consider this correct:

> To determine if we should read from transactions2, we check when usage of transactions2 began, and then read from `_transactions2` if appropriate.